### ISSUE TITLE
DEBUG messages are not visible under systemd

### ISSUE THREAD

**Issue by mzeedijk** [openstreetmap/mod_tile#317]
I use the master branch currently. When renderd is running under systemd DEBUG messages are not logged anymore. I always used the messages that contain "DONE" to quickly see how far my tile process is.
Is there a way to turn them on when running under systemd? I quickly checked the src/g_logger.c and I don't think there is because only ERROR and INFO messages are logged through journald and the rest by default but it's hard to understand what the default is and how to increase the loglevel in case of renderd under systemd.

Also I am not sure if this a bug or an undocumented feature.

**Comment by hummeltech** [comment]
Hello @mzeedijk , you will need to set the environment variable `G_MESSAGES_DEBUG` to `all`:
https://docs.gtk.org/glib/logging.html#debug-message-output

You will have to add something like `Environment="G_MESSAGES_DEBUG=all"` to your systemd service unit.

This is referenced here, but not fully documented:
https://github.com/openstreetmap/mod_tile/blob/master/src/g_logger.c#L85

**Comment by SomeoneElseOSM** [comment]
Re "This is referenced here, but not fully documented", I did add a section to https://switch2osm.org/serving-tiles/manually-building-a-tile-server-ubuntu-22-04-lts/ that says "Making sure that you can see debug messages".  Those instructions are designed for people setting up a tile server for the first time, where more errors are expected.

Like the OP here, I struggled with working out how to turn systemd messages on too.  However, we're in the minority - there were a number of people requesting that debug be turned off by default before that change was merged.

**Comment by cquest** [comment]
Still no luck for me to get DEBUG in syslog (or journalctl) after adding the Environment="G_MESSAGES_DEBUG=all" in the .service file. :(

**Comment by hummeltech** [comment]
Hmm, I'll have to test it out. It should go something like this:

1. Add an override file to the service unit
    ```shell
    sudo systemctl edit renderd.service
    ```
    You will only need to add these two lines:
    ```ini
    [Service]
    Environment=G_MESSAGES_DEBUG=all
    ```
2. Reload the `systemd` manager configuration
    ```shell
    sudo systemctl daemon-reload
    ```
3. Restart the `renderd` service
    ```shell
    sudo systemctl restart renderd.service
    ```
4. Ensure the `override` is effective
    ```shell
    sudo systemctl status renderd.service
    ```
    You should see something like this:
    ```
    Drop-In: /etc/systemd/system/renderd.service.d
             └─override.conf
    ```

**Comment by cquest** [comment]
No change for me, I'm running renderd/focal,now 0.6.1-1~ubuntu20.04.1~ppa1 from osmadmins PPA

**Comment by mzeedijk** [comment]
Same here. I added the Environment line to my service file, but no messages appear in the syslog nor are they visible through journalctl. My service file is now:
```
[Unit]
Description=renderd service
After=postgresql.target

[Service]
Type=forking
User=www-data
RuntimeDirectory=renderd
PIDFile=/var/run/renderd/renderd.pid
ExecStart=/usr/bin/renderd
Restart=always
Environment=G_MESSAGES_DEBUG=all

[Install]
WantedBy=multi-user.target
```

**Comment by SomeoneElseOSM** [comment]
As an example from one that works, here's a bit of debug that I see in /var/log/syslog on https://map.atownsend.org.uk:

    Apr 26 11:19:14 map renderd[707344]: Creating and writing a metatile to /var/cache/renderd/tiles/ajt/14/0/0/0/17/244/237/8.meta

and here is  /usr/lib/systemd/system/renderd.service:

    [Unit]
    Description=Daemon that renders map tiles using mapnik
    Documentation=man:renderd
    After=network.target auditd.service
    
    [Service]
    Environment=G_MESSAGES_DEBUG=all
    ExecStartPre=rm -f /var/cache/renderd/pyosmium.gis/call_pyosmium.running
    ExecStartPre=rm -f /var/cache/renderd/pyosmium.gis3/call_pyosmium.running
    ExecStart=/usr/local/bin/renderd -f
    User=_renderd

    [Install]
    WantedBy=multi-user.target

The "ExecStartPre" entries are related to coordinating data reloads with minutely updates and aren't relevant to this issue.
"renderd -V" returns "0.6.1".  It was built from source from https://github.com/SomeoneElseOSM/mod_tile/tree/zoom2022 , "4 commits ahead, 20 commits behind openstreetmap:master. ".  The "commits ahead" are to do with supporting higher zoom levels and unrelated to this problem.  If there's a genuine issue, it must be related to the "[20 commits behind](https://github.com/SomeoneElseOSM/mod_tile/compare/zoom2022...openstreetmap:mod_tile:master)"

**Comment by mzeedijk** [comment]
It works now on my side. I added "-f" to the ExecStart line and now I see the debug messages in /var/log/syslog.
Also I removed the Type=forking from my service file, because that seemed to restart the service every minute because of a timeout since I introduced the "-f" parameter.

Although I don't see the logic of running a service in "foreground" mode it seems to do the trick. This behavior should be regardless of the way the service is running.

I will close this issue for now. Thank you all for your cooperation.
